lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -160,6 +160,43 @@ class BamDataResult(DataResult):
     max_high: int
 
 
+class DeleteDatasetBatchPayload(BaseModel):
+    src: DatasetSourceType = Field(
+        ...,
+        title="Source",
+        description="The type of dataset associations to delete. Only datasets of this type will be processed.",
+    )
+    ids: List[EncodedDatabaseIdField] = Field(
+        description="The list of datasets IDs to be deleted/purged.",
+    )
+    purge: Optional[bool] = Field(
+        default=False,
+        description="Whether to permanently delete from disk the specified datasets."
+    )
+
+
+class DatasetErrorMessage(BaseModel):
+    dataset_id: EncodedDatabaseIdField = Field(
```

Comment:
A general remark here: Most of the time (but not all, unfortunately) when we write `dataset_id` we are referring to the `id` column of the `dataset` table. "Datasets" as they appear in histories or libraries are always `{History,LibraryDataset}DatasetAssociation`s, which have a `dataset_id` column that points to the id of the Dataset instance, and in some API endpoints we expose this dataset_id. So if we say `dataset_id` here it isn't clear what this refers to.

My suggestion would be to call this field `id` and to add `src` here as well. The client can then translate this into a a more user-friendly error using these two fields.

Reply:
Thanks! I care a lot about correct naming, but it's usually the hardest part :laughing: 
I changed some of the namings in https://github.com/galaxyproject/galaxy/pull/13113/commits/508926f034a6873cf2dc5836938c18ea654e5bbf, but not sure if they are more clarifying. Happy to change them further now that we are at it.

Reply:
That looks good, thank you!